signing/sign.go

```diff
@@ -0,0 +1,96 @@
+package signing
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/ed25519"
```

Comment:
note that ed25519 that is used for spacemesh protocol signatures is not compliant with canonical ed25519 signature

based on the description in the linked issue:
> Verify signature and extract pubkey (as NodeID).

this change should use modified ed25519 instead of canonical

Reply:
It was decided in an offline discussion that the poet will use the custom ed25519 implementation.